spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/JdbcMessageStore.java

```diff
@@ -87,7 +87,7 @@
 				"(GROUP_KEY, REGION, COMPLETE, LAST_RELEASED_SEQUENCE, CREATED_DATE, UPDATED_DATE)"
 				+ " values (?, ?, 0, 0, ?, ?)"),
 
-		UPDATE_MESSAGE_GROUP("UPDATE %PREFIX%MESSAGE_GROUP set UPDATED_DATE=?, CONDITION=? " +
+		UPDATE_MESSAGE_GROUP("UPDATE %PREFIX%MESSAGE_GROUP set UPDATED_DATE=?, GROUP_CONDITION=? " +
```

Comment:
This is a big breaking change in a point release; I think it's ok to change the scripts but I think this code needs to be made conditional on a property (defaulting to CONDITION in the point release and GROUP_CONDITION in 6.0).

Reply:
OK. How about to make it "smart" conditional: based on vendor derived from the provided `DataSource`?
All those which have this `condition` as a reserved word are not going to work anyway.
So, we change the word exactly in their scripts and check for them in the `JdbcMessageStore`.
This way we would leave the rest of supported vendors working as before this change.